src/commands/Register.ts

```diff
@@ -18,130 +18,215 @@ export type RegisterOptions = {
+  const bal = await provider.getBalance(relayAddress);
+
+  if (bal.gt(options.funds)) {
+    log.info('Relayer already funded');
+  } else {
+    log.info('Funding relayer');
+
+    const fundTx = await options.signer.sendTransaction({
+      to: relayAddress,
+      value: options.funds,
+      gasLimit: 1e6,
+      gasPrice: options.gasPrice,
+    });
+
+    if (fundTx.hash == null) {
+      throw new Error(`Fund transaction reverted: ${JSON.stringify(fundTx)}`);
+    }
+    transactions.push(fundTx.hash);
+  }
+
+  await waitForRelay(httpClient, options.relayUrl);
+  log.info('Executed Transactions', transactions);
+};
+
+export async function executeRegister() {
+  // FIXME: add registerOptions?: RegisterOptions from either config or cli (which one?)
   const { app, contracts, blockchain } = getServerConfig();
-  let registerConfig: RegisterConfig | undefined = undefined;
-  if (config.has('register')) {
-    registerConfig = config.get('register');
+  const { account, stake, funds, mnemonic }: RegisterConfig = config.has(
+    'register'
+  )
+    ? config.get<RegisterConfig>('register')
+    : {};
+
+  console.log(
+    `🐞 ᨟ ---------------------------------------------------------------------------------------------------------------------------🐞 ᨟`
+  );
+  console.log(
+    `🐞 ᨟ :: file: register.ts:176 :: executeRegister :: { account, stake, funds, mnemonic }`,
+    { account, stake, funds, mnemonic }
+  );
+  console.log(
+    `🐞 ᨟ ---------------------------------------------------------------------------------------------------------------------------🐞 ᨟`
+  );
+
+  if (account && !mnemonic) {
+    log.error(`
+    You must configure mnemonic for given account address.
+    `);
   }
+
   log.setLevel(app.logLevel);
-  const register = new Register(
-    blockchain.rskNodeUrl,
-    registerConfig?.mnemonic
-  );
+  const rpcProvider = new JsonRpcProvider(blockchain.rskNodeUrl);
   const portIncluded: boolean = app.url.indexOf(':') > 0;
   const relayUrl =
     app.url + (!portIncluded && app.port > 0 ? ':' + app.port.toString() : '');
-  await register.execute(
-    registerOptions
-      ? registerOptions
-      : {
-          hub: contracts.relayHubAddress,
-          signer: await register.findWealthyAccount(),
-          stake: utils.parseEther(registerConfig?.stake ?? '0.01'),
-          funds: utils.parseEther(registerConfig?.funds ?? '0.02'),
-          relayUrl,
-          unstakeDelay: '1000',
-          gasPrice: '60000000',
-        }
+
+  const signer =
+    account && mnemonic
+      ? Wallet.fromMnemonic(mnemonic).connect(rpcProvider)
+      : await findWealthyAccount(rpcProvider);
+
+  const signerAddress = await signer.getAddress();
+  if (signerAddress !== account) {
+    // TODO: probably no point giving the option to configure account if we are going to derive it from the mnemonic. Also should we allow account retreival from private keys (new Wallet(privateKey, rpcProvider))?
```

Comment:
two questions really @antomor :
1. should we allow account config if mnemonic is always necessary anyway?
2. should we allow account retrieval from private keys as well?

Reply:
1. I think that is necessary for validation purposes.
2. We need to accept the private key as-well on my opinion.

Reply:
Ah do you mean to verify that the user didn't make a mistake when passing in the mnemonic?
I can't think of no other verification benefit.

Reply:
@antomor what do you think about this?